build-coordinator/src/main/java/org/jboss/pnc/core/builder/coordinator/bpm/BpmCompleteListener.java

```diff
@@ -16,30 +16,28 @@
  * limitations under the License.
  */
 
-package org.jboss.pnc.core.builder;
+package org.jboss.pnc.core.builder.coordinator.bpm;
 
-import org.jboss.pnc.core.exception.CoreException;
+import org.jboss.pnc.spi.BuildStatus;
 
-import javax.enterprise.inject.Alternative;
-import javax.inject.Inject;
+import javax.enterprise.context.ApplicationScoped;
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * @author <a href="mailto:matejonnet@gmail.com">Matej Lazar</a>
  */
-public class LocalBuildScheduler implements BuildScheduler {
+@ApplicationScoped
+public class BpmCompleteListener {
 
-    BuildExecutor buildExecutor;
+    Map<Long, BpmListener> listeners = new HashMap<>(); //TODO garbage collector after time-out
```

Comment:
We must keep reference, we need to remove them in case BPM server never notifies back.